src/main/java/com/mparticle/kits/ButtonKit.java

```diff
@@ -73,165 +91,222 @@ public ButtonKit getInstance() {
+
     }
 
     @Override
-    public List<ReportingMessage> onActivityStarted(final Activity activity) {
-        Intent intent = activity.getIntent();
-        if (intent == null) {
-            return null;
-        }
-        return handleIntent(intent.getData(), intent.getAction());
+    public void onLoginCompleted(MParticleUser mParticleUser,
+            FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
     }
 
     @Override
-    public List<ReportingMessage> onActivityResumed(final Activity activity) { return null; }
+    public void onLogoutCompleted(MParticleUser mParticleUser,
+            FilteredIdentityApiRequest filteredIdentityApiRequest) {
+        clearAllData();
+    }
 
     @Override
-    public List<ReportingMessage> onActivityPaused(final Activity activity) { return null; }
+    public void onModifyCompleted(MParticleUser mParticleUser,
+            FilteredIdentityApiRequest filteredIdentityApiRequest) {
+
+    }
 
     @Override
-    public List<ReportingMessage> onActivityStopped(final Activity activity) { return null; }
+    public void onUserIdentified(MParticleUser mParticleUser) {
+
+    }
+
+    /*
+     * Overrides for ActivityListener
+     */
 
     @Override
-    public List<ReportingMessage> onActivitySaveInstanceState(final Activity activity, final Bundle outState) { return null; }
+    public List<ReportingMessage> onActivityCreated(Activity activity, Bundle bundle) {
+        trackIncomingIntent(activity.getIntent());
+        return null;
+    }
 
     @Override
-    public List<ReportingMessage> onActivityDestroyed(final Activity activity) { return null; }
+    public List<ReportingMessage> onActivityStarted(Activity activity) {
+        return null;
+    }
 
     @Override
-    public void onApplicationForeground() {
-        checkForAttribution();
+    public List<ReportingMessage> onActivityResumed(Activity activity) {
+        return null;
     }
 
     @Override
-    public void onApplicationBackground() {
+    public List<ReportingMessage> onActivityPaused(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityStopped(Activity activity) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivitySaveInstanceState(Activity activity, Bundle bundle) {
+        return null;
+    }
+
+    @Override
+    public List<ReportingMessage> onActivityDestroyed(Activity activity) {
+        return null;
+    }
+
+    /*
+     * Utility methods
+     */
+
+    private static void logDebug(String message, Object... args) {
+        if (MParticle.getInstance().getEnvironment() == MParticle.Environment.Development) {
+            Logger.debug(String.format("ButtonKit: " + message, args));
+        }
+    }
+
+    private static void logError(String message) {
+        if (MParticle.getInstance().getEnvironment() == MParticle.Environment.Development) {
+            Logger.error("ButtonKit: " + message);
```

Comment:
While it makes sense to only log debug messages in `Development`, shouldn't we also log errors in production?